src/main/java/org/apache/commons/compress/archivers/sevenz/SevenZFile.java

```diff
@@ -629,9 +629,7 @@ private SevenZFile(final Builder builder) throws IOException {
             archive = readHeaders(password);
             this.password = password != null ? Arrays.copyOf(password, password.length) : null;
         } catch (final ArithmeticException | IllegalArgumentException e) {
-            final ArchiveException archiveException = new ArchiveException(e);
-            IOUtils.close(channel, archiveException::addSuppressed);
-            throw archiveException;
+            throw IOUtils.closeQuietly(channel, new ArchiveException(e));
         }
```

Comment:
Honestly, I wouldn't delay the release of `commons-compress` version `1.29.0` more than it already has been to wait for a new version of `commons-io`.

Let's use what we have in `2.21.0`:

```java
        } catch (final Exception e) {
            final IOException ioe = e instanceof IOException ? (IOException) e : new ArchiveException(e);
            IOUtils.closeQuietly(channel, ioe::addSuppressed);
            throw ioe;
        }
```

When we **do** release Commons IO version `2.22.0`, we'll update this call site and **all** the other call sites that might benefit from the new method. There is no chance to forget that, since `closeQuietly` becomes ambiguous in IO `2.22.0`.